src/hotspot/share/oops/constantPool.cpp

```diff
@@ -1421,9 +1421,8 @@ bool ConstantPool::compare_entry_to(int index1, const constantPoolHandle& cp2,
     int k2 = cp2->bootstrap_name_and_type_ref_index_at(index2);
     int i1 = bootstrap_methods_attribute_index(index1);
     int i2 = cp2->bootstrap_methods_attribute_index(index2);
-    // separate statements and variables because CHECK_false is used
-    bool match_entry = compare_entry_to(k1, cp2, k2, CHECK_false);
-    bool match_operand = compare_operand_to(i1, cp2, i2, CHECK_false);
+    bool match_entry = compare_entry_to(k1, cp2, k2);
+    bool match_operand = compare_operand_to(i1, cp2, i2);
     return (match_entry && match_operand);
```

Comment:
Is it worth changing this to:  If (compare_entry_to(...) && compare_operand_to(..)) { .. }
Then if the first one is false the second call isn't needed?

Reply:
I kind of thought these would make a long complicated expression and so the single use variables is helpful.  I don't think performance is important here.

Reply:
-      match = compare_entry_to(recur1, cp2, recur2);
-      if (match) {
+      if (compare_entry_to(recur1, cp2, recur2)) {
         return true;
       }
I could do this

Reply:
You can't return `true` after just `compare_entry_to()` because you still
have to check `compare_operand_to()`.